skbio/sequence/_grammared_sequence.py

```diff
@@ -22,11 +22,79 @@
 from ._sequence import Sequence
 
 
-class IUPACSequence(with_metaclass(ABCMeta, Sequence)):
-    """Store biological sequence data conforming to the IUPAC character set.
+class GrammaredSequenceException(TypeError):
+    pass
+
+
+class GrammaredSequenceMeta(ABCMeta, type):
+    def __new__(mcs, name, bases, dct):
+        cls = super(GrammaredSequenceMeta, mcs).__new__(mcs, name, bases, dct)
+
+        # Only perform validation on classes that aren't abstract.
+        if (type(cls.default_gap_char) is not abstractproperty and
```

Comment:
@jairideout see if this is like what you had in mind. I drew up a dependency graph to try and keep the duplication down and scope properly. Honestly I'm not sure this is worth the extra complexity. If any of these properties are still abstract at instantiation (which for most users will come soon after class declaration time), it's going to fail anyway. I think it would be more straightforward for abstractness to be a class-level distinction. If any member is static, the class should be considered static and the checks shouldn't run. Then once the user gets all the ABC ducks in a row, the static checks will start complaining and they can fix any of those necessary. As it currently stands the user could have ABC errors and static check errors interleaved.